pkg/reconciler/testing/reconcile.go

```diff
@@ -294,18 +313,39 @@ func withoutSink(src v1alpha1.AWSEventSource) {
 }
 
 // Deployed: True
-func deployed(src v1alpha1.AWSEventSource) {
-	src.GetStatus().PropagateAvailability(newReadyDeployment())
+func deployed(adapter runtime.Object) sourceOption {
+	adapter = adapter.DeepCopyObject()
+	ready(adapter)
+
+	return func(src v1alpha1.AWSEventSource) {
+		src.GetStatus().PropagateAvailability(adapter)
+	}
 }
 
 // Deployed: False
-func notDeployed(src v1alpha1.AWSEventSource) {
-	src.GetStatus().PropagateAvailability(newNotReadyDeployment())
+func notDeployed(adapter runtime.Object) sourceOption {
+	adapter = adapter.DeepCopyObject()
+	notReady(adapter)
+
+	return func(src v1alpha1.AWSEventSource) {
+		src.GetStatus().PropagateAvailability(adapter)
+	}
 }
 
 // Deployed: Unknown with error
-func unknownDeployedWithError(src v1alpha1.AWSEventSource) {
-	src.GetStatus().PropagateAvailability(nil)
+func unknownDeployedWithError(adapter runtime.Object) sourceOption {
+	var nilObj runtime.Object
+
+	switch adapter.(type) {
+	case *appsv1.Deployment:
+		nilObj = (*appsv1.Deployment)(nil)
+	case *servingv1.Service:
+		nilObj = (*servingv1.Service)(nil)
+	}
```

Comment:
Here as well. We could do it better if we could somehow access the object's GroupVersionKind. Suggestions welcome.